packages/flutter/lib/src/material/tabs.dart

```diff
@@ -128,7 +136,7 @@ class Tab extends StatelessWidget implements PreferredSizeWidget{
     }
 
     return SizedBox(
-      height: height,
+      height: tabHeight,
```

Comment:
And then just make your check here once.
```suggestion
      height: height ?? calculatedHeight,
```